hosttest/convolutiontest.c

```diff
@@ -0,0 +1,52 @@
+#include <math.h>
+#include <stdbool.h>
+#include <stdio.h>
+
+#include "convolution.h"
+
+#define ARRAY_LEN(_a)	(sizeof(_a)/sizeof(_a[0]))
+#define EPSILON 0.01f
+
+float in[] = {1, 2, 3, 4};
+float filter[] = {-1, 5, 3};
+float out[10];
+
+float expected[10] = {-1.0f, 3.0f, 10.0f, 17.0f, 29.0f, 12.0f, 0.0f, };
```

Comment:
Another test case to compare against is the CMSIS-DSP f32 convolution: https://www.keil.com/pack/doc/CMSIS/DSP/html/arm_convolution_example_f32_8c-example.html